src/nxt_cgroup.c

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (C) Andrew Clayton
+ * Copyright (C) F5, Inc.
+ */
+
+#include <nxt_main.h>
+#include <nxt_process.h>
+#include <nxt_fs.h>
+
+#include <stdio.h>
+#include <string.h>
+#include <limits.h>
+
+
+static nxt_int_t nxt_cgroup_create(const nxt_task_t *task,
+    nxt_process_t *process);
+
+
+static nxt_int_t
+nxt_cgroup_create(const nxt_task_t *task, nxt_process_t *process)
+{
+    char  cgpath[PATH_MAX];
+
+    if (task->thread->runtime->type != NXT_PROCESS_MAIN
+        || nxt_process_type(process) != NXT_PROCESS_PROTOTYPE
+        || process->isolation.cgroup.name == NULL)
+    {
+        return NXT_OK;
+    }
+
+    snprintf(cgpath, sizeof(cgpath), NXT_CGROUP_ROOT "/%s",
+             process->isolation.cgroup.name);
+    nxt_fs_mkdir_all((const u_char *) cgpath, 0777);
+
+    return NXT_OK;
+}
+
+
+nxt_int_t
+nxt_cgroup_proc_add(const nxt_task_t *task, nxt_process_t *process)
+{
+    FILE  *fp;
+    char  cgprocs[PATH_MAX];
+
+    nxt_cgroup_create(task, process);
+
+    if (nxt_process_type(process) != NXT_PROCESS_PROTOTYPE
+        || process->isolation.cgroup.name == NULL)
```

Comment:
This check is repeated in the helper just above?

Reply:
Ah, yes, this harks back to when creating the cgroup and adding the prototype to it were done in different processes, now everything is done from the master process.

In fact I see no point in these being separate functions now, creating the cgroup and adding the prototype process to it happen at the same time.

Reply:
These functions have been combined.